### ISSUE TITLE
PWA vs Native App

### ISSUE THREAD

**Issue by devagx** [TR-SUNSHINE/Frontend#1]
* Identify what features we need to implement and to who e.g. android\ios\both
  * ADV and DISADV of PWA
    * DISADV - Restricted hardware access to IOS
    * DISADV - No push notifications for IOS
    * DISADV - bigger drain on battery power

* Identify based on above if PWA still fits based on what the app needs to deliver, but also taking into account it specifically states on brief “Progressive web app”. 

  * If we feel PWA doesn't fit based on the requirements we need to deliver, then we could argue that we should not deliver a PWA based on reasons x,y,z and also present that in the sprint review presentation.

  * If we feel PWA does match the requirements of what we need to deliver, then think about:

    * Create new react base project based on a PWA template or integrate PWA modules to our existing base project (e.g. add  manifest.json and ServiceWorker files

    * Choose weather to “opt in” or “opt out”

    * Design considerations
      * Notify users that the app works in offline mode
      * Notify users when the service worker has finished getting updates and these updates will be available when they reload the page.

**Comment by Chenerywoman** [comment]
I've put articles on PWAs in the resources file on Google Docs.  I think we could aim to have:
Offline:
- have Weather page with 'home' details, map but I guess no buttons to add reminder, rating etc?  I'm still trying to understand it all but presumably with caching that will be possible
- still not sure the link between caching & local storage - the code-along video doesn't save any data, just has the offline.html

**Comment by Chenerywoman** [comment]
I think we will need a Google Doc with the features we could put in the PWA version

**Comment by Chenerywoman** [comment]
- from an article I'm just reading: "Progressive web apps can send push notifications even when the browser is closed, and the app isn’t active."  However, I've also read that this is only possible for Android phones...
- "Building and maintaining PWAs entails separating static content from dynamic content."
- One of the articles the tutors posted talks about a different create-react-app set up for PWAs.  I guess we stick to our React version for now?  PWA version could just copy content over?
- users visit the website, add the app to a home screen, come back to the home screen, and open the website.
- "So, if native app-specific features (i.e. access to a camera or geolocation) aren’t essential, then stick with a PWA." Not sure if I understand this for our purposes, jf we are going to be using location data?

**Comment by Chenerywoman** [comment]
> I think we will need a Google Doc with the features we could put in the PWA version

I'm not ready to do this yet, as still trying to understand it & what we can manage in timeframe as it is so new to us.

**Comment by devagx** [comment]
ive gone through most of the PWA material also, its quite interesting. And it seems PWA is a hot topic for apps these days.

For location - I think we can make the assumption that "for my location" from the brief will be the location stored in user preferences. I don't think we need to complicate things and implement this, and we could add GPS as a nice to have.  We also know now how to plan a route and save it via google maps, same for obtaining weather based on a location which can be read from the user profile. We just need to explore exactly how we can extract the saved route data to be able to store it (and even if thats possible or not).  I think implementing GPS tracking is way out of the time frame of this project and we can add it as a nice to have feature for future development.

Regarding PWA, as the brief specifically mentions PWA I really want to go PWA, however I think one of the bigger things to look at is push notifications. So far we have been talking about push notifications for reminders etc, but for IOS users we can do that. So if we do go for PWA then we either need to completely rethink what we was originally talking about for push notifications or develop something in specific for ios users to handle it differently which I think will be out of the scope for this project. However, having said that I think if we identify one key piece of functionality such as push notifications not working for PWA for ios users and argue its a key piece of functionality and chose not to implement PWA for that reason, it would show the sponsors we are not just designing and developing something for no real reason.  This also sits along side the agile process of requirements changing throughout the project. Definitely something we need to talk about more in the group during the next sprint as we need to get this topic finalised before we start development.

**Comment by devagx** [comment]
Now im thinking about this more, im doubting the whole topic in general as the comparison for us should be Progressive app VS Normal web app (and not native app as thats an app installed on the phone)), the requirement has always been some kind of web app not mobile app! Some points are still valid which we can take over to the sprint planning discussion

**Comment by Chenerywoman** [comment]
Thanks @devagx I feel that we need to discuss this with the tech coaches to get their advice.  I agree with your last comment - we were never going to do a native app and progressive app is an extension of a web app.

**Comment by devagx** [comment]
moved location discussion in a new ticket
https://github.com/TR-SUNSHINE/Frontend/issues/3

**Comment by devagx** [comment]
as discussed with James. PWA would be more of a stage 2 item to implement. He also mentioned it shouldn't really have been mentioned on our brief and something that he overlooked.
